### ISSUE TITLE
gradio_client fails when full queue

### ISSUE THREAD

**Issue by osanseviero** [gradio-app/gradio#6541]
### Describe the bug

I'm trying to use a Space as an API via `gradio_client`, but I get an error due to full queue (I think) with a `KeyError` message

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```
from gradio_client import Client

client = Client("https://multimodalart-stable-video-diffusion.hf.space/--replicas/z2trf/")
result = client.predict(
		"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",	# filepath  in 'Upload your image' Image component
		0,	# float (numeric value between 0 and 9223372036854775807) in 'Seed' Slider component
		True,	# bool  in 'Randomize seed' Checkbox component
		1,	# float (numeric value between 1 and 255) in 'Motion bucket id' Slider component
		5,	# float (numeric value between 5 and 30) in 'Frames per second' Slider component
							api_name="/video"
)
print(result)
```

### Logs

```shell
Loaded as API: https://multimodalart-stable-video-diffusion.hf.space/--replicas/z2trf/ ✔
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-4-ec496743e655> in <cell line: 4>()
      2 
      3 client = Client("https://multimodalart-stable-video-diffusion.hf.space/--replicas/z2trf/")
----> 4 result = client.predict(
      5                 "https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",     # filepath  in 'Upload your image' Image component
      6                 0,      # float (numeric value between 0 and 9223372036854775807) in 'Seed' Slider component

13 frames
/usr/local/lib/python3.10/dist-packages/gradio_client/client.py in predict(self, api_name, fn_index, *args)
    303                 "Cannot call predict on this function as it may run forever. Use submit instead."
    304             )
--> 305         return self.submit(*args, api_name=api_name, fn_index=fn_index).result()
    306 
    307     def new_helper(self, fn_index: int) -> Communicator:

/usr/local/lib/python3.10/dist-packages/gradio_client/client.py in result(self, timeout)
   1454             >> 9
   1455         """
-> 1456         return super().result(timeout=timeout)
   1457 
   1458     def outputs(self) -> list[tuple | Any]:

/usr/lib/python3.10/concurrent/futures/_base.py in result(self, timeout)
    456                     raise CancelledError()
    457                 elif self._state == FINISHED:
--> 458                     return self.__get_result()
    459                 else:
    460                     raise TimeoutError()

/usr/lib/python3.10/concurrent/futures/_base.py in __get_result(self)
    401         if self._exception:
    402             try:
--> 403                 raise self._exception
    404             finally:
    405                 # Break a reference cycle with the exception in self._exception

/usr/lib/python3.10/concurrent/futures/thread.py in run(self)
     56 
     57         try:
---> 58             result = self.fn(*self.args, **self.kwargs)
     59         except BaseException as exc:
     60             self.future.set_exception(exc)

/usr/local/lib/python3.10/dist-packages/gradio_client/client.py in _inner(*data)
    867             if self.client.serialize:
    868                 data = self.serialize(*data)
--> 869             predictions = _predict(*data)
    870             predictions = self.process_predictions(*predictions)
    871             # Append final output only if not already present

/usr/local/lib/python3.10/dist-packages/gradio_client/client.py in _predict(*data)
    892             }
    893 
--> 894             result = utils.synchronize_async(self._sse_fn, data, hash_data, helper)
    895             if "error" in result:
    896                 raise ValueError(result["error"])

/usr/local/lib/python3.10/dist-packages/gradio_client/utils.py in synchronize_async(func, *args, **kwargs)
    662         **kwargs:
    663     """
--> 664     return fsspec.asyn.sync(fsspec.asyn.get_loop(), func, *args, **kwargs)  # type: ignore
    665 
    666 

/usr/local/lib/python3.10/dist-packages/fsspec/asyn.py in sync(loop, func, timeout, *args, **kwargs)
    104         raise FSTimeoutError from return_result
    105     elif isinstance(return_result, BaseException):
--> 106         raise return_result
    107     else:
    108         return return_result

/usr/local/lib/python3.10/dist-packages/fsspec/asyn.py in _runner(event, coro, result, timeout)
     59         coro = asyncio.wait_for(coro, timeout=timeout)
     60     try:
---> 61         result[0] = await coro
     62     except Exception as ex:
     63         result[0] = ex

/usr/local/lib/python3.10/dist-packages/gradio_client/client.py in _sse_fn(self, data, hash_data, helper)
   1073     async def _sse_fn(self, data: dict, hash_data: dict, helper: Communicator):
   1074         async with httpx.AsyncClient(timeout=httpx.Timeout(timeout=None)) as client:
-> 1075             return await utils.get_pred_from_sse(
   1076                 client,
   1077                 data,

/usr/local/lib/python3.10/dist-packages/gradio_client/utils.py in get_pred_from_sse(client, data, hash_data, helper, sse_url, sse_data_url, cookies)
    339     assert len(done) == 1
    340     for task in done:
--> 341         return task.result()
    342 
    343 

/usr/local/lib/python3.10/dist-packages/gradio_client/utils.py in stream_sse(client, data, hash_data, helper, sse_url, sse_data_url, cookies)
    375                         has_progress = "progress_data" in resp
    376                         status_update = StatusUpdate(
--> 377                             code=Status.msg_to_status(resp["msg"]),
    378                             queue_size=resp.get("queue_size"),
    379                             rank=resp.get("rank", None),

/usr/local/lib/python3.10/dist-packages/gradio_client/utils.py in msg_to_status(msg)
    126     def msg_to_status(msg: str) -> Status:
    127         """Map the raw message from the backend to the status code presented to users."""
--> 128         return {
    129             "send_hash": Status.JOINING_QUEUE,
    130             "queue_full": Status.QUEUE_FULL,

KeyError: 'heartbeat'
```
```


### System Info

```shell
!gradio environment
output
Gradio Environment Information:
------------------------------
Operating System: Linux
gradio version: 4.5.0
gradio_client version: 0.7.0

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
altair: 4.2.2
fastapi: 0.104.1
ffmpy: 0.3.1
gradio-client==0.7.0 is not installed.
httpx: 0.25.1
huggingface-hub: 0.19.4
importlib-resources: 6.1.1
jinja2: 3.1.2
markupsafe: 2.1.3
matplotlib: 3.7.1
numpy: 1.23.5
orjson: 3.9.10
packaging: 23.2
pandas: 1.5.3
pillow: 9.4.0
pydantic: 2.5.1
pydub: 0.25.1
python-multipart: 0.0.6
pyyaml: 6.0.1
requests: 2.31.0
semantic-version: 2.10.0
tomlkit==0.12.0 is not installed.
typer: 0.9.0
typing-extensions: 4.8.0
uvicorn: 0.24.0.post1
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2023.6.0
httpx: 0.25.1
huggingface-hub: 0.19.4
packaging: 23.2
requests: 2.31.0
typing-extensions: 4.8.0
websockets: 11.0.3
```

### Severity

Blocking usage of gradio

**Comment by DUQIA** [comment]
Hello @osanseviero, I have encountered the same situation. Have you solved this error? If so, could you comment on the solution here?

**Comment by freddyaboulton** [comment]
Ok I think this will be fixed by #6693
